Sources/Core/API/TextureManager.swift

```diff
@@ -49,11 +44,18 @@ public final class TextureManager {
                 return nil
             }
 
-            cache[name] = texture
+            cache[cacheKey] = texture
             return texture
         }
 
-        guard let image = imageLoader.loadImageForTexture(named: name, scale: scale) else {
+        let scale = scale ?? defaultScale
+        var name = texture.name
+
+        if let prefix = namePrefix {
+            name = "\(prefix)\(name)"
```

Comment:
Seems like we have some minor code duplication here. Not a big deal, but thinking that you could compute the name before the cache key, then have the method that computes the cache key take a `name`, `format` and `prefix`, instead of a `Texture`? That way you could just compute things once.